sklearn/datasets/tests/test_common.py

```diff
@@ -75,6 +75,14 @@ def check_as_frame(bunch, dataset_func,
     if expected_target_dtype is not None:
         assert np.all(frame_bunch.target.dtypes == expected_target_dtype)
 
+    # Test for return_X_y and as_frame=True
+    frame_X, frame_y = dataset_func(as_frame=True, return_X_y=True)
+    assert isinstance(frame_X, pd.DataFrame)
+    if frame_y.ndim > 1:
+        assert isinstance(frame_X, pd.DataFrame)
+    else:
+        assert isinstance(frame_y, pd.Series)
+
 
 def _skip_network_tests():
     return os.environ.get('SKLEARN_SKIP_NETWORK_TESTS', '1') == '1'
```

Comment:
Shouldn't it be:

```suggestion
    return os.environ.get('SKLEARN_SKIP_NETWORK_TESTS', '0') == '1'
```

by default?

Reply:
No strong opinion, but I was surprise to get those checks skipped by default on my local setup.

Reply:
I think we have no network as default to make it easier for repo maintainers such as debian. (One less configuration to set)

On a side note, it would be nice to have a `[ci network]` to enable this test on the CI for PRs.

Reply:
> No strong opinion, but I was surprise to get those checks skipped by default on my local setup.

I got a little surprised at first as well :)

Reply:
+1 for `[ci network]`. In another PR.